Research/lva-node-app/app.js

```
@@ -0,0 +1,337 @@
+      case 'graphTopologies':
+        graphTopologies=req.body[i].value;
+        break;
+      case 'cameras':
+        cameras=req.body[i].value;
+        break;
+    }
+  }
+}
+
+/**
+* set device ID and IoTHub connection string
+* try to connect to the iot hub to make sure that the passed in values are valid!
+*/
+function setConfigs(req, res)
+{
+  DEVICE_ID=req.body["device-id"];
+  IOTHUB_CONNECTION_STRING=req.body["iothub-connection-string"];
+  convertIotHubToEventHubsConnectionString(IOTHUB_CONNECTION_STRING).then((eventhubString) =>
+  {
+    IOTHUB_ENDPOINT = eventhubString;
+  }).catch((error) =>
+  {
+    console.error(error);
+    res.send(error.message);
+  });
+}
+
+
+/** 
+* broadcast used for sending IoT Hub message data in a live stream 
+* @param {any} data - data to send through websocket
+*/
+wss.broadcast = (data) => 
+{
+    wss.clients.forEach((client) => 
+    {
+      if (client.readyState === WebSocket.OPEN) 
+      {
+        try 
+        {
+          client.send(data);
+        } catch (er) 
+        {
+          console.error(er);
+        }
+      }
+    });
+};
+
+/**
+* function that actually broadcasts hub messages to websocket
+*/
+async function receiveHubMessages() 
+  {
+      const consumerGroup = "$Default"; // name of the default consumer group
+      eventHubReader = new EventHubReader(IOTHUB_CONNECTION_STRING, consumerGroup);
+      (async () => 
+      {
+          await eventHubReader.startReadMessage((message, date, deviceId) => 
+          {
+            try 
+            {
+              const payload = 
+              {
+                IotData: message,
+                MessageDate: date || Date.now().toISOString(),
+                DeviceId: deviceId,
+              };
+
+              wss.broadcast(JSON.stringify(payload));
+            } catch (err) 
+            {
+              console.error('Error broadcasting: [%s] from [%s].', err, message);
+            }
+
+          });
+      })().catch();
```

is there really nothing to do in the empty catch here? if so, add a comment at least when we'd hit this and why we can ignore

---

added in a log statement.